src/glang/lib/std/volatile.c3

```diff
@@ -0,0 +1,71 @@
+typedef [T] Volatile : {
+    underlying: T mut&
+}
+
+function [T] read : (var: Volatile<T>&) -> T = {
+    return volatile_read(&var.underlying);
+}
+
+function [T] write : (var: Volatile<T>&, to_write_data: T) -> void = {
+    volatile_write(&mut var.underlying, to_write_data);
+}
+
+function [T] make<Volatile<T>> : (data: T mut&) -> Volatile<T> = {
+    return {&mut data};
+}
+
+typedef [T] ReadVolatile : {
```

Comment:
What's the use case for this and the other two structs below?

Reply:
For write volatile, I think something like this makes sense:
```graphene
// Memory mapped on the GPU... but also the GPU will never change this value
WriteVolatile<int> screen_scroll = { MAGIC_ADDRESS };

// Tell the GPU to update the scroll... This write MUST happen
screen_scroll:write(20);

// We need to do some math with it (if its already in a register we might as well save time and use that)
let player_position_on_screen : int =  player_position - screen_scroll:read();
```

I'm not sure I can argue for `ReadVolatile` as well.... If it supported member access of large structs you could make an argument someone might DMA into a struct (that you also want to update in-place for some reason). Maybe `ReadOnlyVolatile` makes more sense... it wouldn't have a `write` member.

I added `NonVolatile` just so SFINAE that normally takes volatile types can be written to accept non-volatile types. I'll delete this if I end up changing `ReadVolatile` to `ReadOnlyVolatile`.

Reply:
I've removed `ReadVolatile` and `NonVolatile`